hack/test-cip-image.sh

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# About: This script tests the build process of the cip-image.sh script for both
+# variants (cip and auditor).
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+repo_root=$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd -P)
+
+# Ensures the provided images are built locally.
+validateImages() {
+    local images="${*}"
+    local img_id
+    # Look for every image locally.
+    for img in $images; do
+        # Obtain the image ID.
+        img_id=$(docker image ls "$img" --format '{{.ID}}')
+        # Ensure the image exists.
+        if [[ -z "$img_id" ]]; then
+            >&2 echo "ERROR: Image \"$img\" was not found locally."
+            exit 1
+        fi
+    done
+}
+
+# Validate the container build process for the given variant.
+testVariant() {
+    # Build based on the variant.
+    if [[ "$1" == "auditor" ]]; then
+        make image-build-cip-auditor-e2e
+    else
+        make image-build
+    fi
+
+    # Check required images are built locally.
```

Comment:
Because we have `set -o errexit` on, if the `make` targets above execute successfully, then by definition the images have been built by the time we're here in line 51. So the `validateImages()` isn't providing a lot of value.

It's sort of like saying, "test the compilation of the cip binary by calling `go build` on it and then listing all the local go binaries in the system to see if the binary is there". It's not adding much value.

Anyway, is this script necessary? Why not just make the `test-cip-image-build` target invoke the `image-build` and `image-build-cip-auditor-e2e` targets directly? Isn't that the same thing? You can have multiple things performed in a single make target, like this:

```
some-target:
    doSomething
    doAnotherThing
```

Reply:
This script ensures the required images are built. If we somehow modify or extend cip-image.sh to where the `gcr/k8s-staging-artifact-promoter/cip:[stable-image-tag]` was no longer present, we would catch it with this script. We are essentially guarding the image bundles for cip/prod and auditor/test variants.